build/arduino-esp.yml

```diff
@@ -0,0 +1,105 @@
+      echo $NEWTIN | sudo -S cp -r $ARDUINO_LIBRARY_DIR/AzureIoTHub/examples/iothub_ll_telemetry_sample $ARDUINO_LIBRARY_DIR/../
+      echo $NEWTIN | sudo -S cp $ARDUINO_LIBRARY_DIR/AzureIoTHub/examples/iothub_ll_telemetry_sample/platform.local.esp8266.txt ~/.arduino15/packages/esp8266/hardware/esp8266/2.5.2/platform.local.txt
+    displayName: 'setup libraries'
+    env:
+      IOTHUB_CONNECTION_DEVICE_STRING: $(MXCHIP_NIGHTLY_DEVICE_STRING)
+      WIFI_SSID: $(WIFI_SSID)
+      WIFI_PWD:  $(WIFI_PWD)
+      ARDUINO_LIBRARY_DIR: $(Agent.WorkFolder)/libraries/
+      NEWTIN: $(NEWTIN)
+      
+  - script: |
+      cd $(Agent.WorkFolder)
+      echo "work folder"
+      ls
+      cd libraries/AzureIoTUtility/src
+      echo "util folder"
+      ls
+      cd $(Agent.WorkFolder)/iothub_ll_telemetry_sample
+      echo "telem folder"
+      ls
+      arduino --upload --board esp8266:esp8266:huzzah --pref sketchbook.path=../ --port /dev/ttyUSB0 --verbose-build iothub_ll_telemetry_sample.ino 
+      # should fail job if build/upload fails
+    displayName: 'build and upload'
+    env:
+      ARDUNIO_LIBRARY_DIR: $(Agent.WorkFolder)
+
+  - script: |
+      ls
+      cd $(Build.SourcesDirectory)/testtools/UART_interface/
+      echo -e "exit\r\n\r\n\r\n" | cat > input.txt
+      python3 serial_connect.py -i input.txt -o output.txt -b 115200 -p /dev/ttyUSB0 -d esp8266 -s
+      rm input.txt
+    displayName: 'run_test'
+
+  - script: |
+      # Check that this is the right source directory
+      ls
+      cp $(Build.SourcesDirectory)/testtools/UART_interface/output.txt $(Build.ArtifactStagingDirectory)
+    displayName: "Copy Files into Artifact repo"
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifacts: log'
+    inputs:
+      pathtoPublish: '$(Build.ArtifactStagingDirectory)'
+      artifactName: 'log'
+      publishLocation: 'Container'
+    condition: succeeded()
+
+- job: test_esp32
+  pool:
+    name: devicelab-03
+  steps:
+  - script: |
+      . build_all/arduino/setup_arduino_libraries.sh
+      #copy example and platform.local.txts to their correct locations
+      echo $NEWTIN | sudo -S cp -r $ARDUINO_LIBRARY_DIR/AzureIoTHub/examples/iothub_ll_telemetry_sample $ARDUINO_LIBRARY_DIR/../
+      echo $NEWTIN | sudo -S cp $ARDUINO_LIBRARY_DIR/AzureIoTHub/examples/iothub_ll_telemetry_sample/platform.local.esp32.txt ~/.arduino15/packages/esp32/hardware/esp32/1.0.2/platform.local.txt
+    displayName: 'setup libraries'
+    env:
+      IOTHUB_CONNECTION_DEVICE_STRING: $(MXCHIP_NIGHTLY_DEVICE_STRING)
+      WIFI_SSID: $(WIFI_SSID)
+      WIFI_PWD:  $(WIFI_PWD)
+      ARDUINO_LIBRARY_DIR: $(Agent.WorkFolder)/libraries/
+      NEWTIN: $(NEWTIN)
+
+  - script: |
+      ls
+      cd $(Agent.WorkFolder)/iothub_ll_telemetry_sample/
+      arduino --upload --board esp32:esp32:esp32 --port /dev/ttyUSB1 --pref sketchbook.path=../ --verbose-build iothub_ll_telemetry_sample.ino 
```

Comment:
just out of curiosity what is the likelihood that `/dev/ttyUSB1` changes? Or is this a board that will be plugged into a raspberry pi and never change?

Reply:
It is that, but I should update them to use vault variables anyway.

Reply:
Done.